src/integrations/react.js

```diff
@@ -18,41 +16,34 @@ const CONTEXT_TYPES = {
  *    @connect( state => ({ foo: state.foo, bar: state.bar }) )
  *    export class Foo { render({ foo, bar }) { } }
  */
+
 export function connect(mapStateToProps, actions) {
 	if (typeof mapStateToProps!=='function') {
 		mapStateToProps = select(mapStateToProps || []);
 	}
-	return Child => {
-		function Wrapper(props, context) {
-			Component.call(this, props, context);
-			const store = context.store;
-			let state = mapStateToProps(store ? store.getState() : {}, props);
-			const boundActions = actions ? mapActions(actions, store) : { store };
+	let boundActions;
+	return Child => function Wrapper (props) {
+		const store = useContext(Context);
+		let [state = mapStateToProps(store ? store.getState() : {}, props), setState] = useState();
+		boundActions = boundActions || actions ? mapActions(actions, store) : { store };
+
+		useEffect(() => {
 			let update = () => {
 				let mapped = mapStateToProps(store ? store.getState() : {}, props);
 				for (let i in mapped) if (mapped[i]!==state[i]) {
 					state = mapped;
-					return this.forceUpdate();
+					return setState(state);
 				}
 				for (let i in state) if (!(i in mapped)) {
 					state = mapped;
-					return this.forceUpdate();
+					return setState(state);
 				}
 			};
-			this.componentWillReceiveProps = p => {
-				props = p;
-				update();
-			};
-			this.componentDidMount = () => {
-				store.subscribe(update);
-			};
-			this.componentWillUnmount = () => {
-				store.unsubscribe(update);
-			};
-			this.render = () => createElement(Child, assign(assign(assign({}, boundActions), this.props), state));
-		}
-		Wrapper.contextTypes = CONTEXT_TYPES;
-		return (Wrapper.prototype = Object.create(Component.prototype)).constructor = Wrapper;
+
+			return store.subscribe(update);
+		}, []);
```

Comment:
```suggestion
		}, [store]);
```

Might want to at least put the `store` in here
maybe props as well?